### ISSUE TITLE
Seer not working + Add new bot ?

### ISSUE THREAD

**Issue by Rorolasticot80** [MarlBurroW/RLMarlbot#16]
Hi,
So SeerV4 is not working at all, he just try to score in his own goal while staying on the ground.

And, could you tell us more about adding more bots ourself ?(Like the bot that are on RLBot)

**Comment by MarlBurroW** [comment]
The system I've developed is based on the BaseAgent class from RLBot. The operation is quite straightforward: all RLBot bots have a get_output function that takes a GameTickPacket as a parameter (which contains all the game information for a given frame) and must return a SimpleControllerState (which essentially represents the gamepad inputs).

To add a new bot, you first need to obtain the source code, copy it into your project, and then instantiate the bot (as I do for Nexto, Necto, and Seer). I encourage you to take a look at the code to understand how it's done.

**Comment by Rorolasticot80** [comment]
Yeah i already saw and i tried to do the samething, but when I press the key to turn on the bot, I got this error:


Failed to enable bot:  'NoneType' object is not callable

I did what u said: add the source of the bot (Element in this case), add from element.bot import Element
and this
            elif self.bot_to_use == "element":
                self.bot = Element(player_name, team_index, pri_index)
                self.bot.initialize_agent()

I changed nothing in the Element source code

**Comment by MarlBurroW** [comment]
Try to comment `try:/except:`  blocks to see the entire error stack and get the exact line of the error

**Comment by Rorolasticot80** [comment]
So the error is:
  File "C:\Users\Roro l'asticot\Documents\Hack\last version\rlnexto_python\element\bot.py", line 47, in get_output
    self.game_state.decode(packet, ticks_elapsed)
    ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'decode'

**Comment by MarlBurroW** [comment]
Modify the bot.py line 30 like this:

```python
    def initialize_agent(self, field_info):
        # Initialize the rlgym GameState object now that the game is active and the info is available
        self.game_state = GameState(field_info)
        self.update_action = True
        self.ticks = self.tick_skip  # So we take an action the first tick
        self.prev_time = 0
        self.controls = SimpleControllerState()
        self.action = np.zeros(8)
        self.kickoff_seq = None
```
And give the field_fino when you initialize the bot:

```python
            elif self.bot_to_use == "element":
                self.bot = Element(player_name, team_index, pri_index)
                self.bot.initialize_agent(self.field_info)
                print(Fore.LIGHTGREEN_EX + "Element enabled" + Style.RESET_ALL)
 ```
